### ISSUE TITLE
[Bug] package is deleted when a related function is deleted

### ISSUE THREAD

**Issue by jiangpengcheng** [apache/pulsar#17982]
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.

### Version

db68648c

### Minimal reproduce step

1. upload package

```
❯ ./bin/pulsar-admin  packages upload function://public/default/pfsql-function@0.1.0 --path ~/IdeaProjects/pfsql/pfsql-function/target/pfsql-function-0.1.0-SNAPSHOT.jar --description "pfsql function"
```

2. create function using this package:

```
❯ ./bin/pulsar-admin functions create --name test --output output_topic --inputs input_topic --classname io.streamnative.pfsql.function.FilterRouteFunction --jar function://public/default/pfsql-function@0.1.0
Created successfully
```

3. delete the function:

```
❯ ./bin/pulsar-admin   functions delete --name test
Deleted successfully
```

4. recreate the function, and it report 404 error:

```
❯ ./bin/pulsar-admin functions create --name test2 --output output_topic --inputs input_topic --classname io.streamnative.pfsql.function.FilterRouteFunction --jar function://public/default/pfsql-function@0.1.0

2022-10-10T14:15:35,623+0800 [AsyncHttpClient-7-1] WARN  org.apache.pulsar.client.admin.internal.BaseResource - [http://localhost:8080/admin/v3/functions/public/default/test2] Failed to perform http post request: javax.ws.rs.BadRequestException: HTTP 400 Bad Request
HTTP 404 Not Found

Reason: HTTP 404 Not Found
```

### What did you expect to see?

function recreated successfully

### What did you see instead?

function failed to recreate with 404 error

### Anything else?

the reason is that the builtin "function worker" will delete the related package when a function is deleted:

https://github.com/apache/pulsar/blob/master/pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/rest/api/ComponentImpl.java#L517

I'm not clear about the internal logic, it looks like this is intended behavior while I think this is not reasonable.

In my opinion, a package should  be a sharable one for several different actions and deleting one action should not affect other or new created actions using this package

please correct me if I think it wrong

### Are you willing to submit a PR?

- [x] I'm willing to submit a PR!

**Comment by michaeljmarshall** [comment]
@jiangpengcheng - I think it might make sense to make this a configurable behavior when deleting the function. What do you think?

**Comment by jiangpengcheng** [comment]
@michaeljmarshall 

I think when `worker().getWorkerConfig().isFunctionsWorkerEnablePackageManagement()` is true, it's better to leave the package alone when delete a function

since no matter how packages are created:

1. created manually using package API
2. created automatically when creating a function
3. created by calling `uploadFunction`

they may be used in other places, and deleting them automatically seems not a good idea

how about making users manage such packages manually using package API, What do you think?

**Comment by michaeljmarshall** [comment]
That seems reasonable. I am not familiar enough with the API to have a strong opinion. The main question comes down to classification. If it is a bug, we need to cherry pick it to existing release branches. However, that will introduce a form of breaking change that users might not expect on a patch release. Assuming we do go with your proposed direction, we'll want to make sure that the PR gets labeled with a "release note required" label to warn users of the change.

**Comment by jiangpengcheng** [comment]
this issue is introduced in https://github.com/apache/pulsar/pull/14450, and it only infects branch 2.11(I think there are no releases published for 2.11 yet?)
